include/aws/s3/private/s3_checksums.h

```diff
@@ -26,6 +26,18 @@ struct aws_s3_checksum {
     bool good;
 };
 
+struct aws_s3_checksum_config_storage {
```

Comment:
storage seems a bit weird here, do we use similar naming anywhere else?
can we use options suffix for the checksum config passed in and keep internal as checksum_config?